core/src/main/scala/org/ensime/core/IdleScheduler.scala

```diff
@@ -0,0 +1,28 @@
+package org.ensime.core
+
+import akka.actor.{ Actor, ActorRef, Cancellable }
+import scala.concurrent.duration._
+
+case object Idle
+/*
+Actor that sends Idle  message every times seconds to Project actor.
+ */
+class IdleScheduler(project: ActorRef) extends Actor {
+
+  import context.{ dispatcher, system }
+
+  var idleCallback: Option[Cancellable] = None
+
+  override def preStart(): Unit = {
+    idleCallback = Some(context.system.scheduler.schedule(5.seconds, 5.seconds, project, Idle))
```

Comment:
can you put this in the server config? 5 seconds is way too short as well, probably every 1 minute.